app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,38 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.DiffUtil;
+
+import java.util.Objects;
+
+import ru.yandex.practicum.contacts.presentation.main.ContactUi;
+
+public class BaseListDiffCallback<D extends ListDiffInterface> extends DiffUtil.ItemCallback<D> implements ListDiffInterface<D> {
```

Comment:
Правильной сигнатурой этого класса будет:
```suggestion
public class BaseListDiffCallback <T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
```

Reply:
И соответственно тут нужно будет реализовать 2 метода из класса `DiffUtil.ItemCallback` с вызовом соответствующих методов интерфейса `ListDiffInterface`